WordPress/gutenberg
# Fix default duotone preset SVG and style generation

## Description


This is fixing two issues that both depend on having a new `defaultDuotone` option in theme.json.

Fix #38299 

Prevents the respective styles and SVGs from generating when `defaultPalette`, `defaultGradients`, or `defaultDuotone` are set to `false` in theme.json settings.

In non-fse themes these settings are `false` by default. Colors and gradients have `editor-color-palette` and `editor-gradient-presets` to enable them from #36496.

In fse themes with a theme.json, these settings are `true` by default.

Fix #38690

This makes duotone filters behave more like the colors and gradients.

Enables multi origin colors and duotone presets to be added to the UI.

Adds `defaultDuotone` to disable the default palette from generating.

As a follow-up, we'll want to update the UI to indicate where the presets are coming from like in #35970

## Testing Instructions





In both a classic (non-fse) theme _and_ a block (fse/theme.json) theme with `defaultDuotone` disabled:
  1. View any page/post that doesn't contain a block with duotone applied to it.
  2. See that no duotone SVGs or CSS variables are generated (by searching for "duotone" in the html).

In a block (fse) theme with `defaultDuotone` enabled and theme `duotone` presets included:
  1. Open the duotone palette for a block that supports it.
  2. See both the default and theme duotone presets in the list.

## Screenshots

### Classic Theme (Twenty Twenty)

![](https://user-images.githubusercontent.com/5129775/155021409-da4518af-2c00-4fdf-9873-bc791c0a7648.png)
![](https://user-images.githubusercontent.com/5129775/155021657-cd4404bd-a7fd-4ed8-a9e0-1b46a1570e6a.png)

### Block Theme `"defaultDuotone": true` (Skatepark, default)

![](https://user-images.githubusercontent.com/5129775/155021583-4a2c3436-2ba4-4d5f-b6dc-ade07179d5d3.png)
![](https://user-images.githubusercontent.com/5129775/155021758-4c0735ee-10c1-45f6-aaf6-9167badce7cc.png)

### Block Theme `"defaultDuotone": false` (Skatepark, edited)

![](https://user-images.githubusercontent.com/5129775/155021578-57db2e8a-a5f5-4c9f-a94f-ac10a8374524.png)
![](https://user-images.githubusercontent.com/5129775/155021802-3df88a97-9e91-4921-9e32-4d91dbd980f9.png)

## Types of changes





New feature, bug fix

## Checklist:
